mne/cov.py

```diff
@@ -1129,6 +1129,12 @@ def _get_ch_whitener(A, pca, ch_type, rank):
     """"Get whitener params for a set of channels."""
     # whitening operator
     eig, eigvec = linalg.eigh(A, overwrite_a=True)
+
+    # Clean up our eigenvalues (can be e.g. -1e-26 or 1e-26 for MEG data)
+    eps = np.finfo(float).eps
+    tol = np.max(A.shape) * np.amax(eig) * eps
+    eig[np.abs(eig) < tol] = 0
```

Comment:
Turns out this was the underlying bug via this path:
```
mne.inverse_sparse._gamma_map.gamma_map
mne.inverse_sparse.mxne_inverse._prepare_gain
mne.inverse_sparse.mxne_inverse._prepare_gain_column
mne.minimum_norm.inverse._prepare_forward
mne.cov.prepare_noise_cov
mne.cov._get_ch_whitener
```
For all builds except 2.7 (for whatever reason), the first eigenvalue was -1e-27, so the check in `_prepare_forward` of `nzero = (eig > 0)` correctly removed the first component. But for 2.7, it was 1e26 (slightly positive) so some horrible noise was added to the whitener. Not sure how we've been lucky enough to avoid this previously. I think this is potentially related to the re-application of projectors, but I'm not sure.

This fix uses the same sort of tolerance used by `orth` to determine what the cutoff bound should be. Hopefully it's good enough...